base-server/index.js

```diff
@@ -420,6 +480,10 @@ export class BaseServer {
       channel.regexp = pattern
     }
     this.channels.push(channel)
+
+    let queue = options?.queue || 'main'
+    let channelPattern = channel.regexp ? channel.regexp : channel.pattern.regex
+    this.channelPatternToQueueName.set(channelPattern, queue)
```

Comment:
Can we use `this.channels` instead? Anyway is an array of objects, where we can add new keys.